### DISCUSSION TITLE
HAA to Tasmota

### DISCUSSION THREAD

**Discussion by Jakub1403** [RavenSystem/esp-homekit-devices#1948] [General]
Hi Guys!

I have a Sonoff SV with HAA. Unfortunately, I need to use it with Android phone so I tried to flash Tasmota but with no luck.

Could you please let me know what I'm doing wrong? I have a USB TTL converter, I connected it with Sonoff and I can't make a connection (also I tried to use Arduino Uno to flash Sonoff board - the same). I choose correct COM port and when I click Flash I have info that connection failed.

I think the problem is with flashing mode - when I connect TTL to USB port I'm press and hold button of Sonoff, am I right? Unfortunately, in my case it doesn't do anything. 

How do I put the Sonoff SV with HAA into flashing mode?

Thanks,
Jacob

**Comment by pilot1981** [comment]
Why you don’t publish it to android using Home assistant? It works fine with HAA firmware!

**Reply by Jakub1403** [reply]
I don't use Home assistant right now. I just need to reflash on of my Sonoffs to Tasmota for my friend.

**Reply by mkucharz** [reply]
@Jakub1403 any luck here? I have the same challenge

**Reply by Jakub1403** [reply]
Unfortunately no.

**Reply by mkucharz** [reply]
@Jakub1403 I checked the source code of bootloader (sdk/esp-open-rtos-rsf/bootloader/) and it looks like GPIO 16 has to put in "high" to enter bootloader mode (https://github.com/RavenSystem/esp-homekit-devices/tree/master/sdk/esp-open-rtos-rsf/bootloader/rboot#gpio-boot-mode) . I haven't check that yet.  It is not clear for me which exactly is GPIO 16 in sonoff, but this might be helpful: https://youtu.be/BHzsWwd351I?t=216 (it looks like TX is actually GPIO 16?).
